### PR TITLE
Fix #1854: Support DBT 1.0 format

### LINKED ISSUE #1854: Support DBT 1.0 formats
1. In 0.7 release we added integration however this is for 0.21.0 and 0.21.1
2. But users who upgraded to 1.0 is causing an issue when we do parsing

This issue is to support both 0.21.0 and 1.0 formats.

### PR DESCRIPTION
**Pull request by harshach**
### Describe your changes :
see #1854 

#
### Type of change :

- [x] Bug fix

#
### Frontend Preview (Screenshots) :
<p align="center">For frontend related change, please link screenshots of your changes preview! Optional for backend related changes.
</p>

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [ ] I have performed a self-review of my own. 
- [ ] I have tagged my reviewers below.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#

### CODE REVIEW

File: ingestion/examples/workflows/redshift_dbt.json [hunk 1]
```diff
@@ -2,17 +2,17 @@
   "source": {
     "type": "redshift",
     "config": {
-      "host_port": "cluster.name.region.redshift.amazonaws.com:5439",
-      "username": "username",
-      "password": "strong_password",
+     "host_port": "redshift-cluster-1.clot5cqn1cnb.us-west-2.redshift.amazonaws.com:5439",
+      "username": "awsuser",
+      "password": "focguC-kaqqe5-nepsok",
```
Comment (pmbrull) [inline ingestion/examples/workflows/redshift_dbt.json:7]: is this a real pwd?

File: ingestion/src/metadata/ingestion/source/sql_source.py [hunk 2]
```diff
@@ -358,11 +359,13 @@ def fetch_views(
                 continue
 
     def _parse_data_model(self):
-        logger.info("Parsing Data Models")
+
         if (
             self.config.dbt_manifest_file is not None
             and self.config.dbt_catalog_file is not None
```
Comment (pmbrull) [inline ingestion/src/metadata/ingestion/source/sql_source.py:365]: We could change this for
```python
if (self.config.dbt_manifest_file and self.config.dbt_catalog_file)
```
To use the full `str` truth value and discard empty strings

File: ingestion/src/metadata/ingestion/source/sql_source.py [hunk 3]
```diff
@@ -383,15 +386,21 @@ def _parse_data_model(self):
...
+                compiled_sql = (
+                    mnode["compiled_sql"] if "compiled_sql" in mnode else raw_sql
+                )
+
                 model = DataModel(
                     modelType=ModelType.DBT,
                     description=description,
                     path=path,
-                    rawSql=mnode["raw_sql"] if "raw_sql" in mnode else None,
-                    sql=mnode["compiled_sql"] if "compiled_sql" in mnode else None,
```
Comment (pmbrull) [inline ingestion/src/metadata/ingestion/source/sql_source.py:394]: it actually feels cleaner supporting the `None`, we might want to revisit the `DataModel` processing and have `raw_sql=mnode.get("raw_sql")` and `sql=mnode.get("compiled_sql", raw_sql)`

**Review by pmbrull — APPROVED** [review]
Hi @harshach LGTM, just a couple of convention comments if you got the time or we can directly open a new issue.
Thanks!

**Comment by harshach** [comment]
@pmbrull Thanks for the review. Addressed your comments

**Pull request merged**